geomesa-tools/src/main/scala/org/locationtech/geomesa/tools/ingest/RasterChunking.scala

```diff
@@ -0,0 +1,159 @@
+/*
+ * Copyright 2014 Commonwealth Computer Research, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.locationtech.geomesa.tools.ingest
+
+import java.io.File
+import java.util.UUID
+
+import breeze.numerics.ceil
+import org.geotools.coverage.grid.GridCoverage2D
+import org.locationtech.geomesa.raster.util.RasterUtils.IngestRasterParams
+import org.locationtech.geomesa.utils.geohash.{BoundingBox, GeoHash, GeohashUtils}
+
+import scala.collection.parallel.ForkJoinTaskSupport
+import scala.sys.process._
+import scala.util.Try
+
+class RasterChunking(config: Map[String, Option[String]]) extends RasterIngest {
+  lazy val path = config(IngestRasterParams.FILE_PATH).get
+  lazy val fileType = config(IngestRasterParams.FORMAT).get
+  lazy val parLevel = config(IngestRasterParams.PARLEVEL).get.toInt
+  lazy val chunkSize = config(IngestRasterParams.CHUNKSIZE).get.toInt
+
+  def runChunkTask() = Try {
+    val outDir = s"/tmp/raster_ingest_${UUID.randomUUID.toString}"
+    chunkRasterLocal(path, fileType, outDir, chunkSize)
+    outDir
+  }
+
+  /**
+   * Cut image into chunks so that the size of each chunk is about the given size in bytes,
+   * and save chunks into local geotiff files.
+   *
+   * @param file Raster file to be divided
+   * @param fileType Raster file type
+   * @param outDir Directory to put chunk files in
+   * @param size  Desired size limit in bytes
+   */
+  def chunkRasterLocal(file: String, fileType: String, outDir: String, size: Double) {
+    val rasterReader = getReader(new File(file), fileType)
+    val rasterGrid: GridCoverage2D = rasterReader.read(null)
+    val envelope = rasterGrid.getEnvelope2D
+    val bbox = BoundingBox(envelope.getMinX, envelope.getMaxX, envelope.getMinY, envelope.getMaxY)
+    val (gridStepX, gridStepY) = RasterChunking.getSteps(rasterGrid)
+    val chunkPrec = RasterChunking.GetGeoHashPrecisionBySize(rasterGrid, size)
+    val llGh = GeoHash(envelope.getMinX, envelope.getMinY, chunkPrec)
```

Comment:
We should add code to help deal with input with inputs which may exceed the 'whole world' bounding box.